### PR TITLE
add template function

### PR DESCRIPTION
**Pull request by myaser**
Add template function to be used in k8s-on-aws; it is grouping node-pools by the dedicated label, then finding if there is a common profile, it is then return a string value "karpenter", "zalando", or "" based on the profile

### CODE REVIEW

File: provisioner/template_test.go [hunk 1]
```diff
@@ -1052,3 +1052,115 @@ func TestAWSValidID(t *testing.T) {
...
+						"labels": "dedicated=application-1",
+					},
+				},
+			},
+			expected: map[string]string{
+				"application-1": "zalando",
+			},
+		},
+		{
+			name: "application-1 dedicated pools share the same profile, which is un-known",
```
Comment (gargravarr) [inline provisioner/template_test.go:1085]: un-known -> unknown

**Comment by gargravarr** [comment]
:+1:

**Comment by myaser** [comment]
👍🏻

File: provisioner/template.go [hunk 2]
```diff
@@ -686,13 +698,34 @@ func zoneDistributedNodePoolGroups(nodePools []*api.NodePool) map[string]bool {
 		}
 	}
 
-	result := make(map[string]bool)
+	return poolGroups
+}
+
+// nodeLifeCycleProviderPerNodePoolGroup groups node-pools by the dedicated label, each node-pool-group is mapped to
+// the provider suitable for its node-pool profile
+// in case all pools of a group do not share the same profile, the group is mapped to empty string
```
Comment (mikkeloscar) [inline provisioner/template.go:706]: What does this mean in practice? Which provider would be used here by admission-controller? :thinking:

Comment (myaser) [reply provisioner/template.go:706]: an empty string "" provider will be used, which is practically turning this feature off

**Pull request merged**